pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -182,3 +182,13 @@ def init_optimizers(self, trainer: "Trainer", model: LightningModule):
 
     def optimizer_step(self, optimizer: torch.optim.Optimizer, lambda_closure: Callable, **kwargs):
         optimizer.step(closure=lambda_closure, **kwargs)
+
+    @property
+    def setup_optimizers_in_pre_dispatch(self) -> bool:
+        """
+        Override to delay setting optimizers and schedulers till after dispatch.
+        This is useful when the `TrainingTypePlugin` requires operating on the wrapped accelerator model.
+        However this may break certain precision plugins such as APEX which require optimizers to be set.
```

Comment:
n00b question: why does this break APEX?

Reply:
APEX wraps the optimizer into an FP16_Optimizer (I've sure you've seen this in a few places). The optimizer needs to be defined for this to work :(

Annoying because this is the only reason for this overcomplication within the training type plugin.